### PR TITLE
Add GH Issue Template

### PR DESCRIPTION
**Pull request by anishrajan25**
Fixes Issue #4675

### CODE REVIEW

**Comment by anishrajan25** [comment]
@claudio4j Could you please review the PR?

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  - type: markdown
+    attributes:
+      value: |
+        ## Camel K setup
+        Please describe your Camel K setup.
+  - type: input
+    id: version
+    attributes:
+      label: Camel K version
+      description: Which version of the Camel K are you running?
```
Comment (claudio4j) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:76]: Add something like "Verify if the kamel cli is the same version as the camel-k-operator with: kamel version -a"

**Review by claudio4j — CHANGES_REQUESTED** [review]
Hey Anish, this template looks very good, I just requested some minor changes.
Thank you for the contribution.

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    validations:
+      required: true
+  - type: textarea
+    id: logs
+    attributes:
+      label: Relevant log output
+      description: |
+        Please copy and paste any relevant log output.
+        This will be automatically formatted into code, so no need for backticks.
+        Please indicate from which component the logs originate.
```
[the remark this answers is not in the document]
Comment (anishrajan25) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:55]: Sure, I have added additional pointers to add these helper points

**Comment by anishrajan25** [comment]
Here are the screenshots for reference as to how the templates would look once rendered:
### Issues Page
image

---

### Bug Report Page
image

---

### Feature Request Page
image

**Review by squakez — COMMENTED** [review]
IMO there's too much information required for any issue. If we want to collect information it would be enough to provide some text form with the version, the log and a free from with the description. My concern is that we are adding a barrier for anybody willing to submit a feedback. 

It also is making references to CNCF instead of Apache foundation.

For me it's a -1, but if the rest of the community is okey with it, at least, let's make it simpler.

cc @oscerd @lburgazzoli @gansheer @christophd

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+labels: ["kind/bug"]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Welcome to Apache Camel K! 👋🎉 Thanks for taking the time to fill out this bug report.
+
+        Before you post:
+        - Please search for existing issues to avoid creating duplicate bugs/feature requests.
+        - Please be respectful and considerate of others (see [CNCF Code of Conduct](https://github.com/cncf/foundation/blob/main/code-of-conduct.md)).
```
Comment (claudio4j) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:13]: Can you this link for the code of conduct 
https://www.apache.org/foundation/policies/conduct

**Comment by claudio4j** [comment]
Well, let's try to reduce some extra fields, Anish wdyt to have only one textarea field for the following subjects ?
What happened?
Steps to reproduce
Expected behavior

And I think we can drop these fields
Screenshot
Additional context
Language
Operating system

It's going to be good for a first iteration of this form.

**Comment by christophd** [comment]
In my experience people ignore the given fields most of the time. But I like the pointers to zulip chat and mailing list as part of the landing page for new issues. This may direct people to a better suited discussion forum for their individual question.

Also +1 to combining at least `What happened?` and `Expected behavior` fields into one single question. In my experience people describe the whole scenario in `What happened?` section and then write the exact same statement in slightly other wording to the `Expected behavior` field (like What happended? "This and that did not work properly for me", Expected behavior: "This and that should work properly")

**Comment by anishrajan25** [comment]
@claudio4j I am aligned on dropping the following fields:
- Screenshot
- Additional context
- Language
- Operating system

Also as @christophd suggested, it would be good to merge `What happened?` and `Expected behavior` fields into one.

But I would suggested to have the `Steps to Reproduce` and `Relevant log output` as separate fields in addition to the `What happened?` because:
- As per my experience (and I'm a beginner when it comes to Open Source), at times the issues don't have much information and it gets hard for someone who is fairly new to the project to understand what exactly is the issue.
- Having  `Steps to Reproduce` and `Relevant log output` would definitely help in providing additional information to folks who want to contribute.
- Also having the field for `Camel-K version` would also be a helpful thing.
- I understand these fields might act as a barrier in some cases, so we can mark the fields other than `What happened?` as optional.

**Comment by claudio4j** [comment]
Sounds good Anish, my understanding is is already good for a first try, I will approve it and we can get back to it later if it requires changes to allow a more friendly page.
Thank you Anish.

**Comment by anishrajan25** [comment]
I have updated the PR as discussed. @claudio4j You can have a final look.

This is how the Bug Report Form would look like now:
image

**Review by claudio4j — APPROVED** [review]

**Pull request merged**